crates/astria-composer/src/searcher/executor.rs

```diff
@@ -0,0 +1,321 @@
+            }
+            Ok(Err(e)) => {
+                error!(
+                    error.message = %e,
+                    error.cause_chain = ?e,
+                    "executor task failed unexpectedly with error",
+                );
+            }
+            Err(e) => {
+                error!(
+                    error.message = %e,
+                    error.cause_chain = ?e,
+                    "executor task panicked",
+                );
+            }
+        }
+    });
+
+    Ok((executor_tx, status_rx))
+}
+
+#[derive(Debug)]
+pub(super) struct Executor {
+    // The status of this executor
+    status: watch::Sender<Status>,
+    // Channel for receving bundles to pack, sign, and submit
+    executor_rx: mpsc::Receiver<Vec<Action>>,
+    // The client for submitting wrapped and signed pending eth transactions to the astria
+    // sequencer.
+    sequencer_client: SequencerClient,
+    // Private key used to sign sequencer transactions
+    sequencer_key: SigningKey,
+    // Nonce of the sequencer account we sign with
+    nonce: Arc<AtomicU32>,
+    // The sequencer address associated with the private key
+    address: Address,
+}
+
+#[derive(Debug)]
+pub(super) struct Status {
+    is_connected: bool,
+}
+
+impl Status {
+    pub(super) fn new() -> Self {
+        Self {
+            is_connected: false,
+        }
+    }
+
+    pub(super) fn is_connected(&self) -> bool {
+        self.is_connected
+    }
+}
+
+/// The `Executor` interfaces with the sequencer. It handles account nonces, transaction signing,
+/// and transaction submission.
+/// The `Executor` receives `Vec<Action>` from the bundling logic, packages them with a nonce into
+/// an `Unsigned`, then signs them with the sequencer key and submits to the sequencer.
+/// Its `status` field indicates that connection to the sequencer node has been established.
+impl Executor {
+    pub(super) fn new(
+        sequencer_url: &str,
+        private_key: &SecretString,
+        executor_rx: mpsc::Receiver<Vec<Action>>,
+    ) -> eyre::Result<Self> {
+        // connect to sequencer node
+        let sequencer_client =
+            SequencerClient::new(sequencer_url).wrap_err("failed constructing sequencer client")?;
+
+        // create signing key for sequencer txs
+        let mut private_key_bytes: [u8; 32] = hex::decode(private_key.expose_secret())
+            .wrap_err("failed to decode private key bytes from hex string")?
+            .try_into()
+            .map_err(|_| eyre!("invalid private key length; must be 32 bytes"))?;
+        let sequencer_key =
+            SigningKey::try_from(private_key_bytes).wrap_err("failed to parse sequencer key")?;
+        private_key_bytes.zeroize();
+
+        // create address from signing key
+        let sequencer_address = Address::from_verification_key(sequencer_key.verification_key());
+
+        // create channel for status reporting
+        let (status, _) = watch::channel(Status::new());
```

Comment:
where's the receiver? the receiver has to be used otherwise sending on this channel will error due to all receivers being dropped